curves/bn254/g2_conversions.go

```diff
@@ -1,14 +1,18 @@
+//go:build g2
+
 package bn254
 
 import (
-	"github.com/consensys/gnark-crypto/ecc/bn254"
-	"github.com/consensys/gnark-crypto/ecc/bn254/fp"
-	icicle "github.com/ingonyama-zk/icicle/goicicle/curves/bn254"
 	"fmt"
+
+	bn254 "github.com/consensys/gnark-crypto/ecc/bn254"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fp"
+	"github.com/ingonyama-zk/icicle/wrappers/golang/core"
+	icicle_bn254 "github.com/ingonyama-zk/icicle/wrappers/golang/curves/bn254"
 )
 
-func ToGnarkFp(f *icicle.G2Element) *fp.Element {
-	fb := f.ToBytesLe()
+func ToGnarkFp(f *icicle_bn254.G2BaseField) *fp.Element {
+	fb := f.ToBytesLittleEndian()
 	var b32 [32]byte
 	copy(b32[:], fb[:32])
```

Comment:
We should use fp.Bytes here instead of hardcoded values for the size of the array